record/record.go

```diff
@@ -158,9 +160,17 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 			r.i = r.j + headerSize
```

Comment:
I merged the block of code above with a call to recover, as the previous behavior was somewhat similar and now it reads cleaner imho.